File: crowdControl/src/main/java/com/example/crowdControl/models/Visit.java
package com.example.crowdControl.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Data
public class Visit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int visitId;

    @ManyToOne
    private Visitor visitor;

    @ManyToOne
    private Shop shop;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateTimeIn;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateTimeOut;

}

File: angular/src/app/models/Visit.ts
import { IShop, Shop } from './shop';
import { IVisitor } from './Visitor';

export interface IVisit {
    visitId: number,
    shop: IShop,
    visitor: IVisitor,
    dateTimeIn?: Date,
    dateTimeOut?: Date
}

export class Visit implements IVisit {
    visitId: number;
    shop: IShop;
    visitor: IVisitor;
    dateTimeIn: Date;
    dateTimeOut: Date;
}
